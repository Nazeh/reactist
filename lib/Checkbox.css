.reactist_checkbox {
  font-size: 0.875rem;
  color: #202020;
  font-weight: 400;
  line-height: 1.7;
}
.reactist_checkbox:hover {
  cursor: pointer;
}
.reactist_checkbox .reactist_checkbox--input {
  margin-right: 10px;
}

