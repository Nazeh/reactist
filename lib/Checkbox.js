!function(e,t){"object"==typeof exports&&"object"==typeof module?module.exports=t(require("prop-types"),require("react")):"function"==typeof define&&define.amd?define("reactist",["prop-types","react"],t):"object"==typeof exports?exports.reactist=t(require("prop-types"),require("react")):e.reactist=t(e["prop-types"],e.React)}(window,function(e,t){return function(e){var t={};function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}};return e[r].call(o.exports,o,o.exports,n),o.l=!0,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,"a",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p="",n(n.s=21)}([function(t,n){t.exports=e},function(e,n){e.exports=t},,function(e,t){e.exports=function(e){var t=[];return t.toString=function(){return this.map(function(t){var n=function(e,t){var n=e[1]||"",r=e[3];if(!r)return n;if(t&&"function"==typeof btoa){var o=function(e){return"/*# sourceMappingURL=data:application/json;charset=utf-8;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(e))))+" */"}(r),i=r.sources.map(function(e){return"/*# sourceURL="+r.sourceRoot+e+" */"});return[n].concat(i).concat([o]).join("\n")}return[n].join("\n")}(t,e);return t[2]?"@media "+t[2]+"{"+n+"}":n}).join("")},t.i=function(e,n){"string"==typeof e&&(e=[[null,e,""]]);for(var r={},o=0;o<this.length;o++){var i=this[o][0];"number"==typeof i&&(r[i]=!0)}for(o=0;o<e.length;o++){var a=e[o];"number"==typeof a[0]&&r[a[0]]||(n&&!a[2]?a[2]=n:n&&(a[2]="("+a[2]+") and ("+n+")"),t.push(a))}},t}},function(e,t,n){var r={},o=function(e){var t;return function(){return void 0===t&&(t=e.apply(this,arguments)),t}}(function(){return window&&document&&document.all&&!window.atob}),i=function(e){var t={};return function(e,n){if("function"==typeof e)return e();if(void 0===t[e]){var r=function(e,t){return t?t.querySelector(e):document.querySelector(e)}.call(this,e,n);if(window.HTMLIFrameElement&&r instanceof window.HTMLIFrameElement)try{r=r.contentDocument.head}catch(e){r=null}t[e]=r}return t[e]}}(),a=null,c=0,s=[],u=n(5);function f(e,t){for(var n=0;n<e.length;n++){var o=e[n],i=r[o.id];if(i){i.refs++;for(var a=0;a<i.parts.length;a++)i.parts[a](o.parts[a]);for(;a<o.parts.length;a++)i.parts.push(b(o.parts[a],t))}else{var c=[];for(a=0;a<o.parts.length;a++)c.push(b(o.parts[a],t));r[o.id]={id:o.id,refs:1,parts:c}}}}function l(e,t){for(var n=[],r={},o=0;o<e.length;o++){var i=e[o],a=t.base?i[0]+t.base:i[0],c={css:i[1],media:i[2],sourceMap:i[3]};r[a]?r[a].parts.push(c):n.push(r[a]={id:a,parts:[c]})}return n}function p(e,t){var n=i(e.insertInto);if(!n)throw new Error("Couldn't find a style target. This probably means that the value for the 'insertInto' parameter is invalid.");var r=s[s.length-1];if("top"===e.insertAt)r?r.nextSibling?n.insertBefore(t,r.nextSibling):n.appendChild(t):n.insertBefore(t,n.firstChild),s.push(t);else if("bottom"===e.insertAt)n.appendChild(t);else{if("object"!=typeof e.insertAt||!e.insertAt.before)throw new Error("[Style Loader]\n\n Invalid value for parameter 'insertAt' ('options.insertAt') found.\n Must be 'top', 'bottom', or Object.\n (https://github.com/webpack-contrib/style-loader#insertat)\n");var o=i(e.insertAt.before,n);n.insertBefore(t,o)}}function d(e){if(null===e.parentNode)return!1;e.parentNode.removeChild(e);var t=s.indexOf(e);t>=0&&s.splice(t,1)}function h(e){var t=document.createElement("style");if(void 0===e.attrs.type&&(e.attrs.type="text/css"),void 0===e.attrs.nonce){var r=function(){0;return n.nc}();r&&(e.attrs.nonce=r)}return v(t,e.attrs),p(e,t),t}function v(e,t){Object.keys(t).forEach(function(n){e.setAttribute(n,t[n])})}function b(e,t){var n,r,o,i;if(t.transform&&e.css){if(!(i=t.transform(e.css)))return function(){};e.css=i}if(t.singleton){var s=c++;n=a||(a=h(t)),r=m.bind(null,n,s,!1),o=m.bind(null,n,s,!0)}else e.sourceMap&&"function"==typeof URL&&"function"==typeof URL.createObjectURL&&"function"==typeof URL.revokeObjectURL&&"function"==typeof Blob&&"function"==typeof btoa?(n=function(e){var t=document.createElement("link");return void 0===e.attrs.type&&(e.attrs.type="text/css"),e.attrs.rel="stylesheet",v(t,e.attrs),p(e,t),t}(t),r=function(e,t,n){var r=n.css,o=n.sourceMap,i=void 0===t.convertToAbsoluteUrls&&o;(t.convertToAbsoluteUrls||i)&&(r=u(r));o&&(r+="\n/*# sourceMappingURL=data:application/json;base64,"+btoa(unescape(encodeURIComponent(JSON.stringify(o))))+" */");var a=new Blob([r],{type:"text/css"}),c=e.href;e.href=URL.createObjectURL(a),c&&URL.revokeObjectURL(c)}.bind(null,n,t),o=function(){d(n),n.href&&URL.revokeObjectURL(n.href)}):(n=h(t),r=function(e,t){var n=t.css,r=t.media;r&&e.setAttribute("media",r);if(e.styleSheet)e.styleSheet.cssText=n;else{for(;e.firstChild;)e.removeChild(e.firstChild);e.appendChild(document.createTextNode(n))}}.bind(null,n),o=function(){d(n)});return r(e),function(t){if(t){if(t.css===e.css&&t.media===e.media&&t.sourceMap===e.sourceMap)return;r(e=t)}else o()}}e.exports=function(e,t){if("undefined"!=typeof DEBUG&&DEBUG&&"object"!=typeof document)throw new Error("The style-loader cannot be used in a non-browser environment");(t=t||{}).attrs="object"==typeof t.attrs?t.attrs:{},t.singleton||"boolean"==typeof t.singleton||(t.singleton=o()),t.insertInto||(t.insertInto="head"),t.insertAt||(t.insertAt="bottom");var n=l(e,t);return f(n,t),function(e){for(var o=[],i=0;i<n.length;i++){var a=n[i];(c=r[a.id]).refs--,o.push(c)}e&&f(l(e,t),t);for(i=0;i<o.length;i++){var c;if(0===(c=o[i]).refs){for(var s=0;s<c.parts.length;s++)c.parts[s]();delete r[c.id]}}}};var y=function(){var e=[];return function(t,n){return e[t]=n,e.filter(Boolean).join("\n")}}();function m(e,t,n,r){var o=n?"":r.css;if(e.styleSheet)e.styleSheet.cssText=y(t,o);else{var i=document.createTextNode(o),a=e.childNodes;a[t]&&e.removeChild(a[t]),a.length?e.insertBefore(i,a[t]):e.appendChild(i)}}},function(e,t){e.exports=function(e){var t="undefined"!=typeof window&&window.location;if(!t)throw new Error("fixUrls requires window.location");if(!e||"string"!=typeof e)return e;var n=t.protocol+"//"+t.host,r=n+t.pathname.replace(/\/[^\/]*$/,"/");return e.replace(/url\s*\(((?:[^)(]|\((?:[^)(]+|\([^)(]*\))*\))*)\)/gi,function(e,t){var o,i=t.trim().replace(/^"(.*)"$/,function(e,t){return t}).replace(/^'(.*)'$/,function(e,t){return t});return/^(#|data:|http:\/\/|https:\/\/|file:\/\/\/|\s*$)/i.test(i)?e:(o=0===i.indexOf("//")?i:0===i.indexOf("/")?n+i:r+i.replace(/^\.\//,""),"url("+JSON.stringify(o)+")")})}},,,,,,,,,,,,,,,,function(e,t,n){"use strict";n.r(t);n(22);var r=n(1),o=n.n(r),i=n(0),a=n.n(i),c=function(e){var t=e.label,n=e.checked,r=e.onChange;return o.a.createElement("label",{className:"reactist checkbox"},o.a.createElement("input",{className:"checkbox--input",value:t,checked:n,onChange:function(e){return r(e.target.checked)},type:"checkbox"}),t)};c.displayName="Checkbox",c.defaultProps={checked:!1},c.propTypes={onChange:a.a.func.isRequired,checked:a.a.bool,label:a.a.string},t.default=c},function(e,t,n){var r=n(23);"string"==typeof r&&(r=[[e.i,r,""]]);var o={hmr:!0,transform:void 0,insertInto:void 0};n(4)(r,o);r.locals&&(e.exports=r.locals)},function(e,t,n){(e.exports=n(3)(!1)).push([e.i,".reactist.checkbox {\n  font-size: 0.875rem;\n  color: #202020;\n  font-weight: 400;\n  line-height: 1.7;\n}\n.reactist.checkbox:hover {\n  cursor: pointer;\n}\n.reactist.checkbox .checkbox--input {\n  margin-right: 10px;\n}\n",""])}])});
//# sourceMappingURL=Checkbox.map