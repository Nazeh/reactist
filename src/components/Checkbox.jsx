import './styles/checkbox.less'

import React from 'react'
import PropTypes from 'prop-types'

const Checkbox = ({ label, checked, onChange }) => (
    <label className="reactist checkbox">
        <input
            className="checkbox--input"
            value={label}
            checked={checked}
            onChange={event => onChange(event.target.checked)}
            type="checkbox"
        />
        {label}
    </label>
)
Checkbox.displayName = 'Checkbox'
Checkbox.defaultProps = {
    checked: false
}
Checkbox.propTypes = {
    /** Handler function that is called when the checkbox is toggled. Is invoked with the checked value and not the full event. */
    onChange: PropTypes.func.isRequired,
    /** Current value of the checkbox. */
    checked: PropTypes.bool,
    /** Label that is displayed next to the checkbox. */
    label: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ])
}

export default Checkbox
