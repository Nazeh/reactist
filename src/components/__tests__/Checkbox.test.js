import React from 'react'
import { shallow } from 'enzyme'

import Checkbox from '../Checkbox'

describe('Checkbox', () => {
    it('renders when initially set to false', () => {
        const checkbox = shallow(getCheckbox())
        expect(checkbox).toMatchSnapshot()
    })

    it('renders when initially set to true', () => {
        const checkbox = shallow(getCheckbox({ checked: true }))
        expect(checkbox).toMatchSnapshot()
    })

    it('calls onChange callback when checkbox value changes', () => {
        const onChangeSpy = jest.fn()
        const checkbox = shallow(getCheckbox({ onChange: onChangeSpy }))

        checkbox
            .find('.checkbox--input')
            .simulate('change', { target: { checked: true } })
        expect(onChangeSpy).toHaveBeenLastCalledWith(true)
    })

    // Helpers ================================================================
    const getCheckbox = props => (
        <Checkbox
            label="Checkbox Label"
            checked={false}
            onChange={jest.fn()}
            {...props}
        />
    )
})
