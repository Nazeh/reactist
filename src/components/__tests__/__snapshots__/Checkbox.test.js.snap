// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders when initially set to false 1`] = `
<label
  className="reactist_checkbox"
>
  <input
    checked={false}
    className="checkbox--input"
    onChange={[Function]}
    type="checkbox"
    value="Checkbox Label"
  />
  Checkbox Label
</label>
`;

exports[`Checkbox renders when initially set to true 1`] = `
<label
  className="reactist_checkbox"
>
  <input
    checked={true}
    className="checkbox--input"
    onChange={[Function]}
    type="checkbox"
    value="Checkbox Label"
  />
  Checkbox Label
</label>
`;
